### PR TITLE
Adding unit tests for /Cart/Data.php

### PR DESCRIPTION
**Pull request by njfaries**
# Description
Added unit tests for /Cart/Data.php. 

There are three lines not quite covered that I will be getting coverage for later as my lack of familiarity with m2 means I'm not sure how to get it covered. Given the delays in this project already though 93% is good enough to merge and tackle other tests.

Fixes: https://app.asana.com/0/1131034199022718/1137538625478186/f

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: Test/Unit/Controller/Cart/DataTest.php [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+     * @var RequestInterface request
+     */
+    private $request;
+
+    /**
+     * @var MetricsClient metricsClient
+     */
+    private $metricsClient;
+
+    private $response;
```
Comment (matt-thomason) [inline Test/Unit/Controller/Cart/DataTest.php]: Missing typehinting on these variables

File: Test/Unit/Controller/Cart/DataTest.php [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+        $this->response = (object) ( array(
+            'cart' =>
+                (object) ( array(
+                    'order_reference' => self::ORDER_REFERENCE,
+                    'display_id'      => '100050001 / 1234',
+                    'shipments'       =>
+                        array(
+                            0 =>
+                                (object) ( array(
+                                    'shipping_addres' => $this->requestShippingAddress,
```
Comment (matt-thomason) [inline Test/Unit/Controller/Cart/DataTest.php]: Typo in `shipping_address`

File: Test/Unit/Controller/Cart/DataTest.php [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+                                            'currency_symbol' => '$',
+                                        ) ),
+                                    'reference'       => 'flatrate_flatrate'
+                                ) ),
+                        ),
+                ) ),
+            'token' => self::TOKEN
+        ) );
+
+        $this->responseData = json_decode(json_encode($this->response), true);
```
Comment (matt-thomason) [inline Test/Unit/Controller/Cart/DataTest.php]: I know this was copied from the actual code, but a short comment explaining why this is done would be good.

**Comment by matt-thomason** [comment]
Can you rebase your PR to catch @daisy1754 's fix for circleci?

**Comment by njfaries** [comment]
Rebased

**Pull request merged**